pageserver/src/consumption_metrics.rs

```diff
@@ -32,7 +33,37 @@ struct Ids {
     timeline_id: Option<TimelineId>,
 }
 
-/// Key that uniquely identifies the object, this metric describes.
+/// Name of the metric, used by `MetricsKey` factory methods and `deserialize_cached_events`
+/// instead of static str.
+// Do not rename any of these without first consulting with data team and partner
+// management.
+// FIXME: write those tests before refactoring to this!
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+enum Name {
+    /// Timeline last_record_lsn, absolute
+    #[serde(rename = "written_size")]
```

Comment:
Serde has a mode to rename to snake case: https://serde.rs/container-attrs.html#rename_all

Maybe you could use that one here? Some of the variants would have to be renamed but otherwise it would save us these attributes.

Reply:
No, I want to keep these manual, though... Now I added tests.

Reply:
> Note not all would work with rename_all..

Yeah that's what I meant with renaming, the internal CamelCase name would have to be renamed.

We can always do this later, let's focus on getting the feature shipped to the customer, so resolving this concern.